### PR TITLE
#160195763 Api endpoints to create a comment and get comments

### PR DESCRIPTION
**Pull request by d3mola**
#### What does this PR do?
Creates api endpoints to create a comment and get comments

#### Description of Task to be completed?
- Create Comments model and migrations
- Associate Requests to Comments by a one-to-many relationship
- Create controller method to create a comment
- Modify `getRequestDetails` controller so that it eager-loads comments associated with it
- Create middleware to validate comments query
- Write tests for the server side
- Seed Requests table before running Comments controller tests
- Decode auth token based on the node environment
- Create route and controller to create a new travel request

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Change into the repo directory - `cd travel_tool_back`
- Navigate to [travela](http://travela-staging.andela.com/) in a browser
- Open the developer tools, locate applications tab, copy `jwt-token` in the cookies
- Install dependencies and start the server - `make start`
- Open postman; under `Headers`, create a field called `Authorization` and paste the value of token copied from the browser
- Still in postman, send a POST request to `localhost:5000/api/v1/requests` to create a new request
- Find information about the fields for the above the `POST` request [here](https://github.com/andela/travel_tool_back/pull/14)
- From the response of the created request, copy the value of `requestId`
- Still in postman, send a post request to `localhost:5000/api/v1/comments` with following fields
    - `comment` - the comment you want to post
    - `requestId` - the requestId you copied
- You should have created a new comment successfully.
- To get the list of comments for a request, in postman, make a `GET` request to `localhost:5000/api/v1/requests/requestId`, where requestId is the id of the request you created earlier.

#### Any background context you want to provide?
A request has to exist before a comment can be created, that is why we had to go through the trouble of creating a request before creating a comment
[Run in Postman](https://app.getpostman.com/run-collection/5180493385e56b2601cc)

#### What are the relevant pivotal tracker stories?
[#160195763](https://www.pivotaltracker.com/story/show/160195763)

#### Screenshots (if appropriate)
image

#### Questions:
None

### CODE REVIEW

**Comment by Spectrumsun** [comment]
Good Job. Please add an endpoint to get the list of comments made to the PR description @d3mola

File: src/modules/comments/CommentsController.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  static async createComment(req, res) {
+    try {
+      const { requestId } = req.body;
+      const { name, email, picture } = req.user.UserInfo;
+      const commentData = {
+        ...req.body, userName: name, userEmail: email, picture,
+      };
+      const request = await models.Request.findById(requestId);
+      // only create a comment if the request exists
+      if (requestId === (request && request.id)) {
```
Comment (amarachukwu-agbo) [inline src/modules/comments/CommentsController.js]: Please refactor the conditional statement here. I think 'if (requestId) will suffice'

Comment (d3mola) [reply src/modules/comments/CommentsController.js]: @amarachukwu-agbo done.

**Comment by babadee001** [comment]
LGTM @samuelkubai

**Pull request merged**